hrm-domain/hrm-core/permissions/actions.ts

```diff
@@ -48,6 +55,17 @@ export const actionsMaps = {
     ADD_CONTACT_TO_CASE: 'addContactToCase',
     REMOVE_CONTACT_FROM_CASE: 'removeContactFromCase',
   },
+  profile: {
+    VIEW_PROFILE: 'viewProfile',
+    // EDIT_PROFILE: 'editProfile', // we don't need edit for now, will be needed when users can attach more identifiers or edit the name
+    ASSOCIATE_PROFILE_TO_PROFILE_FLAG: 'associateProfileToProfileFlag',
+    DISASSOCIATE_PROFILE_TO_PROFILE_FLAG: 'disassociateProfileToProfileFlag',
+  },
+  profileSection: {
```

Comment:
Are we 100% sure that having a separate profileSection Kind is the best way to go here? I think it's a very likely scenario that there will be some requirement in the future for a condition that relates to the parent profile. For example 'I only want to users who own a contact associated with the profile to be able to create new sections'

I know we could achieve this by always pulling the parent profile as well as the section for the purposes of the permission check, but might we end up duplicating a lot of conditions across the 'profile' and 'profileSection' kinds in that way? Just seeing that currently our 'create new profile section' permission checks are checked against a null entity because we have nothing to provide to the check troubles me. I just don't want to paint ourselves into a corner.

Reply:
I think the same applies to the case sections, and we found that it was worst to represent those as part of the parent case because then you have to do lot of weird logic just to grant the granular control expected - e.g. "can edit the section only if the type is note". We would need a `canEditProfileNote` which seems just what we are trying to move away from in cases to me :thinking:.  Also I just don't see how this prevents the issue of pulling the profile and the profile section anyways - we do need the profile section for sure to determine it's type, is not just about the top level profile.
Maybe I'm missing some cases that the current representation would make harder?

As a more abstract thought:
If we need to add more information about the "parent entity", or even about the "graph of all the related entities", I think we could start using the recently introduced notion of `context` to the permissions system, to for example, fetch the profile, the contacts related to the profile, etc **only if** that's defined in the conditions sets, otherwise we don't need the extra context and we are just evaluating properties about the section. What do you think?

PS: We moved away from defining permissions rules in JS because it made no sense to have that much expressiveness, but if we start considering supporting everything that could be expressed in our business model, then trying to do so with JSON based predicates seems hard to maintain in the long run, maybe we want to consider going back to TS defined rules?